tests/resources/serializers/test_ui_serializer.py

```diff
@@ -9,14 +9,38 @@
 """Resources serializers tests."""
 
 import json
+from copy import deepcopy
+
+import pytest
 
 from invenio_rdm_records.resources.serializers import UIJSONSerializer
 
 
-def test_ui_serializer(app, full_record_ui):
-    """Test UI serializer."""
-    # full_record_ui = full_record
-    full_record_ui['access']['status'] = 'embargoed'
+@pytest.fixture(scope='function')
+def full_to_dict_record(full_record):
```

Comment:
I would still call it `ui` record... it is more clear in the fact that uses the UI serialization (some sort of).... the full_record is already a dict 😅

Reply:
Here is how I understand things / the context behind the naming (maybe I got it wrong @lnielsen):   
 
We have all these layers now (data layer, service layer, UI layer, API layer). But each layer has its own representation of a record. This is good because we isolate layers and their responsibilities. Unfortunately we are in the mix of things, so not everything is as clear cut (yet!).

| Layer   | Representation                                                 |
| ------- | -------------------------------------------------------------- |
| UI   (top)   | UIJSONSerializer().serialize_object_to_dict(result_item.to_dict())  --> UIRecord    |
| API     | result_item.to_dict()  --> APIRecord (`full_to_dict_record`)                          |
| Service | ResultItem  ---> ServiceRecord                                 |
| Data  (bottom)   | Essentially Record.create() --> DataRecord                     |

`UIJSONSerializer().serialize_object_to_dict(result_item.to_dict())` returns a UIRecord as opposed to taking one as input. The `full_record` fixture is meant to represent the dict coming from the JSON API originally. But it probably got bastardized along the way :sweat_smile:. I think the naming scheme above would make things clearer if adopted but making these changes is hard :shrug: .